# Results comparison for MIST with fine-tuned models of model A partial

In this notebook, I will compare the results of the different fine tuned models (only the MIST dataset).

In [7]:
from ML.utils.utils import *
from physics.Iso_data_handler import Iso_data_handler
from ML.utils.Data_preparator import Data_preparator

In [8]:
pre_path = "../../../../../../"
path = pre_path + "results/model_A_partial/fine_tuned_models"

## Getting the ranges for the values

In [3]:
iso_handler = Iso_data_handler("C:/Users/antoi/Code/unif/MA2/thesis/data/MIST_v1.2_vvcrit0.0_basic_isos/", 
                                    ['log10_isochrone_age_yr', 'log_Teff', 'log_g', 'phase', 'metallicity', 'star_mass', 'log_R', 'log_L'], 
                                    "MIST", reclassify=True)

iso_df = iso_handler.get_isochrone_dataframe()
display(iso_df)

Reading MIST dataframe from MIST_reclassified_iso_full_data.csv file...


,log10_isochrone_age_yr,log_Teff,log_g,phase,metallicity,star_mass,log_R,log_L
0,5.0,3.486221,3.131342,-1.0,-0.25,0.100000,0.153402,-0.793667
1,5.0,3.487362,3.126808,-1.0,-0.25,0.102645,0.160326,-0.775254
2,5.0,3.489243,3.119367,-1.0,-0.25,0.107039,0.171785,-0.744810
3,5.0,3.491102,3.112165,-1.0,-0.25,0.111419,0.183099,-0.714749
4,5.0,3.492937,3.105143,-1.0,-0.25,0.115789,0.194305,-0.684996
...,...,...,...,...,...,...,...,...
1467117,10.3,4.402490,7.777159,6.0,0.50,0.532726,-1.806255,-1.047903
1467118,10.3,4.387132,7.783242,6.0,0.50,0.532730,-1.809295,-1.115415
1467119,10.3,4.371789,7.789130,6.0,0.50,0.532735,-1.812237,-1.182672
1467120,10.3,4.356480,7.794844,6.0,0.50,0.532741,-1.815091,-1.249617


In [4]:
phase_filtered_iso_df = Data_preparator.filter_data(iso_df, {'phase':[0, 2, 3], 'star_mass':("<", 30)})
print_all_uniques(phase_filtered_iso_df)

log10_isochrone_age_yr  : Range : 5.0 - 10.3, Mean : 8.2913, Median : 8.3

log_Teff  : Range : 3.34 - 5.3542, Mean : 3.9814, Median : 3.9163

log_g  : Range : -1.08 - 6.2434, Mean : 3.5356, Median : 3.9078

Values in phase column : 0.0, 2.0, 3.0 

Values in metallicity column : -0.25, -0.5, -0.75, -1.0, -1.25, -1.5, -1.75, -2.0, -2.5, -3.0, -3.5, -4.0, 0.0, 0.25, 0.5 

star_mass  : Range : 0.1 - 29.9991, Mean : 5.0421, Median : 2.422

log_R  : Range : -1.0 - 3.1298, Mean : 0.6593, Median : 0.6146

log_L  : Range : -3.33 - 5.9763, Mean : 2.1988, Median : 2.3277



## Comparison

| Parameter      | mass_filtered_smaller_14                                           |
| -------------  | ------------------------------------------------------------------ |
| age            | Range : 5.0 - 10.3, Mean : 8.4693, Median : 8.5                    |
| mass           | star_mass  : Range : 0.1 - 13.9995, Mean : 3.3831, Median : 2.0289 |
| log_teff       | log_Teff  : Range : 3.34 - 5.2138, Mean : 3.9383, Median : 3.867   |
| log_g          | log_g  : Range : -1.08 - 7.9405, Mean : 3.5574, Median : 3.9537    |
| log_R          | log_R  : Range : -1.86 - 3.0828, Mean : 0.6007, Median : 0.5357    |

### Mass parameter

In [10]:
compare_metrics(path, value_rounding=3,
                output_parameters=["mass"],
                physical_models=["MIST"],)

Results for the Global category with a value of 1.0


,,,,value range,RVE,RMSE,MAE,MedAE,CORR,MAX_ER,Percentiles,time
model,physical_model,filter,output_parameter,,,,,,,,,
decision_tree,MIST,mass_smaller_14,mass,0.1 - 13.999,1.0,0.051,0.016,0.003,1.0,5.054,75 : 0.01204 / 90 : 0.03835 / 95 : 0.07323 / 99 : 0.22204,17.916
KNN,MIST,mass_smaller_14,mass,0.1 - 13.999,1.0,0.058,0.020,0.005,1.0,2.562,75 : 0.01728 / 90 : 0.0509 / 95 : 0.08791 / 99 : 0.2393,11.112
random_forest,MIST,mass_smaller_14,mass,0.1 - 13.999,1.0,0.031,0.011,0.003,1.0,1.852,75 : 0.00931 / 90 : 0.02399 / 95 : 0.04372 / 99 : 0.13158,1986.565


Results for the phase category with a value of 0.0


,,,,value range,RVE,RMSE,MAE,MedAE,CORR,MAX_ER,Percentiles,time
model,physical_model,filter,output_parameter,,,,,,,,,
decision_tree,MIST,mass_smaller_14,mass,0.1 - 13.999,1.0,0.051,0.017,0.004,1.0,5.054,75 : 0.01362 / 90 : 0.04204 / 95 : 0.07397 / 99 : 0.19481,17.916
KNN,MIST,mass_smaller_14,mass,0.1 - 13.999,1.0,0.063,0.021,0.006,1.0,2.562,75 : 0.0181 / 90 : 0.04796 / 95 : 0.0786 / 99 : 0.2504,11.112
random_forest,MIST,mass_smaller_14,mass,0.1 - 13.999,1.0,0.025,0.007,0.002,1.0,1.852,75 : 0.00611 / 90 : 0.01564 / 95 : 0.02668 / 99 : 0.07487,1986.565


Results for the phase category with a value of 2.0


,,,,value range,RVE,RMSE,MAE,MedAE,CORR,MAX_ER,Percentiles,time
model,physical_model,filter,output_parameter,,,,,,,,,
decision_tree,MIST,mass_smaller_14,mass,0.654 - 13.928,1.0,0.025,0.008,0.000,1.0,1.411,75 : 0.00578 / 90 : 0.02328 / 95 : 0.03465 / 99 : 0.08609,17.916
KNN,MIST,mass_smaller_14,mass,0.654 - 13.928,1.0,0.031,0.012,0.003,1.0,0.643,75 : 0.01036 / 90 : 0.03343 / 95 : 0.06162 / 99 : 0.13788,11.112
random_forest,MIST,mass_smaller_14,mass,0.654 - 13.928,1.0,0.019,0.009,0.005,1.0,0.463,75 : 0.01036 / 90 : 0.02108 / 95 : 0.03297 / 99 : 0.08395,1986.565


Results for the phase category with a value of 3.0


,,,,value range,RVE,RMSE,MAE,MedAE,CORR,MAX_ER,Percentiles,time
model,physical_model,filter,output_parameter,,,,,,,,,
decision_tree,MIST,mass_smaller_14,mass,0.545 - 13.994,0.999,0.074,0.027,0.003,1.0,2.379,75 : 0.01668 / 90 : 0.07299 / 95 : 0.14939 / 99 : 0.34725,17.916
KNN,MIST,mass_smaller_14,mass,0.545 - 13.994,1.000,0.072,0.031,0.006,1.0,1.351,75 : 0.02859 / 90 : 0.09174 / 95 : 0.15361 / 99 : 0.3117,11.112
random_forest,MIST,mass_smaller_14,mass,0.545 - 13.994,1.000,0.053,0.024,0.007,1.0,0.713,75 : 0.02273 / 90 : 0.0662 / 95 : 0.10988 / 99 : 0.23824,1986.565


### Radius parameter

In [11]:
compare_metrics(path, value_rounding=3,
                output_parameters=["radius"],
                physical_models=["MIST"],)

Results for the Global category with a value of 1.0


,,,,value range,RVE,RMSE,MAE,MedAE,CORR,MAX_ER,Percentiles,time
model,physical_model,filter,output_parameter,,,,,,,,,
decision_tree,MIST,mass_smaller_14,radius,-0.997 - 3.082,1.0,0.009,0.005,0.003,1.0,0.446,75 : 0.00675 / 90 : 0.01225 / 95 : 0.01613 / 99 : 0.03482,17.916
KNN,MIST,mass_smaller_14,radius,-0.997 - 3.082,1.0,0.005,0.003,0.002,1.0,0.179,75 : 0.00346 / 90 : 0.00661 / 95 : 0.00922 / 99 : 0.01696,11.112
random_forest,MIST,mass_smaller_14,radius,-0.997 - 3.082,1.0,0.004,0.002,0.001,1.0,0.117,75 : 0.00227 / 90 : 0.00472 / 95 : 0.00715 / 99 : 0.01498,1986.565


Results for the phase category with a value of 0.0


,,,,value range,RVE,RMSE,MAE,MedAE,CORR,MAX_ER,Percentiles,time
model,physical_model,filter,output_parameter,,,,,,,,,
decision_tree,MIST,mass_smaller_14,radius,-0.997 - 3.082,1.0,0.006,0.004,0.002,1.0,0.311,75 : 0.00456 / 90 : 0.00807 / 95 : 0.01132 / 99 : 0.02404,17.916
KNN,MIST,mass_smaller_14,radius,-0.997 - 3.082,1.0,0.004,0.003,0.002,1.0,0.057,75 : 0.00279 / 90 : 0.00559 / 95 : 0.00795 / 99 : 0.01523,11.112
random_forest,MIST,mass_smaller_14,radius,-0.997 - 3.082,1.0,0.002,0.001,0.001,1.0,0.045,75 : 0.00115 / 90 : 0.00231 / 95 : 0.00342 / 99 : 0.00636,1986.565


Results for the phase category with a value of 2.0


,,,,value range,RVE,RMSE,MAE,MedAE,CORR,MAX_ER,Percentiles,time
model,physical_model,filter,output_parameter,,,,,,,,,
decision_tree,MIST,mass_smaller_14,radius,0.091 - 2.871,0.999,0.011,0.007,0.006,1.0,0.446,75 : 0.00984 / 90 : 0.01384 / 95 : 0.01724 / 99 : 0.03213,17.916
KNN,MIST,mass_smaller_14,radius,0.091 - 2.871,1.000,0.004,0.003,0.002,1.0,0.064,75 : 0.00419 / 90 : 0.00709 / 95 : 0.00906 / 99 : 0.01321,11.112
random_forest,MIST,mass_smaller_14,radius,0.091 - 2.871,1.000,0.003,0.002,0.002,1.0,0.046,75 : 0.00318 / 90 : 0.00548 / 95 : 0.00736 / 99 : 0.01169,1986.565


Results for the phase category with a value of 3.0


,,,,value range,RVE,RMSE,MAE,MedAE,CORR,MAX_ER,Percentiles,time
model,physical_model,filter,output_parameter,,,,,,,,,
decision_tree,MIST,mass_smaller_14,radius,0.076 - 3.064,0.999,0.014,0.007,0.004,1.0,0.212,75 : 0.00865 / 90 : 0.01704 / 95 : 0.02612 / 99 : 0.05948,17.916
KNN,MIST,mass_smaller_14,radius,0.076 - 3.064,1.000,0.008,0.004,0.002,1.0,0.179,75 : 0.00438 / 90 : 0.00871 / 95 : 0.01276 / 99 : 0.03767,11.112
random_forest,MIST,mass_smaller_14,radius,0.076 - 3.064,1.000,0.008,0.004,0.002,1.0,0.117,75 : 0.00507 / 90 : 0.01006 / 95 : 0.01438 / 99 : 0.0308,1986.565


Random forest is the best model from all the fine tuned ones